core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerProxy.scala

```diff
@@ -770,6 +770,31 @@ object ContainerProxy {
           causedBy ++ initTime ++ binding
       })
   }
+
+  /**
+   * Partitions the activation arguments into two JsObject instances. The first is exported as intended for export
+   * by the action runtime to the environment. The second is passed on as arguments to the action.
+   *
+   * @param content the activation arguments
+   * @param initArgs set of parameters to treat as initialization arguments
+   * @return A partition of the arguments into an environment variables map and the JsObject argument to the action
+   */
+  def partitionArguments(content: Option[JsObject], initArgs: Set[String]): (Map[String, JsString], JsObject) = {
+    content match {
+      case None                         => (Map.empty, JsObject.empty)
+      case Some(js) if initArgs.isEmpty => (Map.empty, js)
+      case Some(js) =>
+        val (env, args) = js.fields.partition(k => initArgs.contains(k._1))
+
+        (env.map {
```

Comment:
Just a minor concern - This logic would be executed for both `/init` and `/run` flow while its only required for `/init`. If we can avoid this for `/run` (which is  the more critical path) then that would reduce unnecessary object creations..

May be break partition into 2 parts - partition and stringify and have stringify logic invoked only for `/init` flows

Reply:
Agreed. I wanted to defer stringify to `containerInitializer().`

Reply:
https://github.com/apache/openwhisk/pull/4559/commits/1fb74dcccd834846975eaf68883875853d03b834 relocated the conversion to the initializer.